docs/src/modules/concepts/pages/ai-agents.adoc

```diff
@@ -0,0 +1,62 @@
+= AI Agents
+
+include::ROOT:partial$include.adoc[]
+
+AI agents are components that integrate with AI to perceive their environment, make decisions, and take actions toward a specific goal. Agents can have varying degrees of human intervention from none (completely autonomous) to requiring a human to approve each action the agent takes.
+
+== Tokens and streaming
+
+Agents interact with AI, most commonly in the form of Large Language Models (LLMs). LLMs are what is known as _predictive text_. This means that every word streamed to the agent is actually just the next word predicted to be in the output. Regardless of platform or language, agents need the ability to stream tokens bi-directionally.
+
+If your agent is consuming an LLM as a service, then you could be paying some amount of money per bundle of tokens. In cases like this, it is crucial to ensure that you have control over how frequently and how many tokens the agent "spends."
+
+== Different types of AI
+
+LLMs are everywhere these days and it is impossible to escape all of their related news. It would be easy to assume that all agents interact with LLMs whether they are self-hosted or provided as an external service. This idea does a disservice to the rest of machine learning and AI in particular. 
+
+As you develop your teams of collaborative agents, keep in mind that not everything needs to be an LLM and look for opportunities to use smaller, more efficient, task-specific models. This can not only save you money, but can improve the overall performance of your application.
+
+== Prompts, session memory, and context
```

Comment:
This section is more the core of an agent, and would be a good starting point for the page.